biggus/__init__.py

```diff
@@ -955,7 +955,7 @@ def __init__(self, array, axes=None):
             axes = np.arange(array.ndim)[::-1]
         elif len(axes) != array.ndim:
             raise ValueError('Incorrect number of dimensions.')
-        self.axes = list(axes)
+        self.axes = axes
```

Comment:
I guess self.axes should be private really. Equally, the only place where an assumption that self.axes is mutable is in a single function - it is there that the assumption must be ratified.

P.S. What are you doing working on biggus at this time of the night! :wink: